src/gateway/WebSocket.ts

```diff
@@ -3,39 +3,52 @@ import WebSocket from 'ws';
 import { EVENTS, Payload } from './constants';
 import { GATEWAY_OPCODES, GATEWAY_CLOSE_EVENT_CODES } from '../util/Constants';
 
-export default function Socket(token: string, wsURL: string, intents: number) {
-  let interval: NodeJS.Timeout;
-  let lastHeartbeatAck = true;
-  let session_id: string;
-  let seq: number;
-  let ws: WebSocket;
-
-  function heartbeat(expected: boolean) {
+export default class Socket {
```

Comment:
This should at least have the same naming as the file name.
I personally would call that Client. since we are in gateway folder.
We could also go with WebSocketClient, DiscordWebSocket, etc...
It depends what we ultimately want:
- this class will emits node events for every discord events
- this class is given an instance of EventEmitter when constructed. The class emitting events will be this class.

I suggest the second solution as it seems to better separate responsabilities. 
In this case we should probably go with `DiscordWebSocket` (concrete class to handle websocket connection with discord).

Reply:
we'll probably do the EventEmitter part in the next pull request
changed the names btw, used DiscordWebSocket